From e7972a3a23ff1b5f9b7b11eda5c9b5c0c16b8787 Mon Sep 17 00:00:00 2001
From: Jesse Chan <jc@lineageos.org>
Date: Thu, 18 Oct 2018 18:21:35 -0700
Subject: [PATCH 22/24] HWComposer: HWC2: allow SkipValidate to be force
 disabled

HWC on some devices does not support SkipValidate.
On earlier versions, hasCapability(HWC2::Capability::SkipValidate)
will be called to check if SkipValidate is available.

This check has been removed in Pie and now surfaceflinger
rely on HWC to fallback to validate when there is any client layer.
But HWC is not reliable as some vendors do not implement fallback at all.

Thus, this change adds a flag to allow SkipValidate to be force disabled.

Change-Id: I61f69bd82cad74820e42246182c3d5b131137831
---
 build/soong/android/variable.go | 4 ++++
 build/soong/soong_config.mk     | 1 +
 2 files changed, 5 insertions(+)

diff --git a/build/soong/android/variable.go b/build/soong/android/variable.go
index cf04bc7e..f578a347 100644
--- a/build/soong/android/variable.go
+++ b/build/soong/android/variable.go
@@ -20,6 +20,9 @@ type Product_variables struct {
 	Has_legacy_camera_hal1 struct {
 		Cflags []string
 	}
+	Hwc2_no_skipvalidate struct {
+		Cppflags []string
+	}
 	Target_process_sdk_version_override struct {
 		Cppflags []string
 	}
@@ -48,6 +51,7 @@ type ProductVariables struct {
 	Device_support_legacy_hwfde  *bool `json:",omitempty"`
 	Device_support_wait_for_qsee  *bool `json:",omitempty"`
 	Has_legacy_camera_hal1  *bool `json:",omitempty"`
+	Hwc2_no_skipvalidate  *bool `json:",omitempty"`
 	Java_Source_Overlays *string `json:",omitempty"`
 	Specific_camera_parameter_library  *string `json:",omitempty"`
 	Target_process_sdk_version_override *string `json:",omitempty"`
diff --git a/build/soong/soong_config.mk b/build/soong/soong_config.mk
index ae518cca..da717730 100644
--- a/build/soong/soong_config.mk
+++ b/build/soong/soong_config.mk
@@ -9,6 +9,7 @@ $(call add_json_bool, Device_support_hwfde_perf, $(filter true,$(TARGET_HW_DISK_
 $(call add_json_bool, Device_support_legacy_hwfde, $(filter true,$(TARGET_LEGACY_HW_DISK_ENCRYPTION)))
 $(call add_json_bool, Device_support_wait_for_qsee, $(filter true,$(TARGET_KEYMASTER_WAIT_FOR_QSEE)))
 $(call add_json_bool, Has_legacy_camera_hal1, $(filter true,$(TARGET_HAS_LEGACY_CAMERA_HAL1)))
+$(call add_json_bool, Hwc2_no_skipvalidate, $(filter true,$(TARGET_HWC2_NO_SKIPVALIDATE)))
 $(call add_json_str, Java_Source_Overlays, $(JAVA_SOURCE_OVERLAYS))
 $(call add_json_str, Specific_camera_parameter_library, $(TARGET_SPECIFIC_CAMERA_PARAMETER_LIBRARY))
 $(call add_json_str_omitempty, Target_process_sdk_version_override, $(TARGET_PROCESS_SDK_VERSION_OVERRIDE))
-- 
2.17.1

